eos-chrome-plugin-update-upstream

```diff
@@ -34,104 +34,119 @@ WIDEVINE_VERSION=
+}
+
 fp_exit() {
    echo $1
    rm -fr ${TEMP_DIR}
    exit 0
 }
 
-fp_exit_with_error() {
-   echo $1
-   echo "The Chromium plugins are NOT installed."
-   [ -d ${DEST_DIR} ] || mkdir -p ${DEST_DIR}
-   touch ${WIDEVINE_VERSION_FILE}
-   rm -fr ${TEMP_DIR}
-   exit 1
+fp_get_field() {
+    grep-dctrl -n -s $1 ${CHROME_FLAVOUR_VERSION} ${TEMP_DIR}/Packages
 }
 
-fp_exit_retry_with_error() {
-   echo $1
-   echo "The Chromium plugins are NOT installed."
-   rm -fr ${TEMP_DIR}
-   exit 1
-}
+fp_should_check_updates_widevine() {
+    if [ ! -f ${WIDEVINE_VERSION_FILE} ] ; then
+        WIDEVINE_VERSION="not installed"
+    else
+        WIDEVINE_VERSION=$(cat ${WIDEVINE_VERSION_FILE})
+        if ! [ `find ${WIDEVINE_VERSION_FILE} -daystart -mtime +${DAYS_CHECK}` ] && \
+           ! [ `find ${WIDEVINE_VERSION_FILE} -newermt "0 days"` ] ; then
+            # Has been checked recently. No need to check
+            echo "No need to check the Widevine plugin yet (version: ${WIDEVINE_VERSION})"
+            return 1
+        fi
+    fi
 
-fp_get_field() {
-    grep-dctrl -n -s $1 ${CHROME_FLAVOUR_VERSION} ${TEMP_DIR}/Packages
+    echo "Widevine plugin is not installed or might be too old. Checking for a new version"
+    return 0
 }
 
-fp_find_latest_version() {
-    ${WGET_COMMAND} "${CHROME_PACKAGES}" -O "${TEMP_DIR}/Packages" \
-        || fp_exit_retry_with_error "Failed to download Packages from Google Chrome"
+fp_fetch_widevine_metadata() {
+    if ! ${WGET_COMMAND} "${CHROME_PACKAGES}" -O "${TEMP_DIR}/Packages"; then
+        echo "Failed to download Packages from Google Chrome"
+        return 1
+    fi
+
     local package=$(fp_get_field "Package")
-    [ -n "${package}" ] || \
-        fp_exit_with_error "Packages does not contain ${CHROME_FLAVOUR_VERSION}"
+    if [ -z "${package}" ]; then
+        echo "Packages does not contain ${CHROME_FLAVOUR_VERSION}"
+        return 1
+    fi
+
     CHROME_VERSION=$(fp_get_field "Version")
     CHROME_SHA1SUM=$(fp_get_field "SHA1")
+
     local filename=$(fp_get_field "Filename")
     CHROME_URL_FILENAME=${CHROME_URL}/${filename}
     CHROME_FILENAME=${CHROME_URL_FILENAME##*/}
 }
 
-fp_should_check_update() {
-    if [ ! -f ${WIDEVINE_VERSION_FILE} ] ; then
-        WIDEVINE_VERSION="not installed"
-    elif [ `find ${WIDEVINE_VERSION_FILE} -daystart -mtime +${DAYS_CHECK}` ] || \
-        [ `find ${WIDEVINE_VERSION_FILE} -newermt "0 days"` ] ; then
-        WIDEVINE_VERSION=$(cat ${WIDEVINE_VERSION_FILE})
+fp_should_update_widevine() {
+    if fp_fetch_widevine_metadata; then
+       if [ "${WIDEVINE_VERSION}" = "${CHROME_VERSION}" ] ; then
+           # Touch the VERSION so it is not checked again until ${DAYS_CHECK} passed
+           echo "Widevine plugin is up-to-date"
+           touch ${WIDEVINE_VERSION_FILE}
+           return 1
+       else
+           echo "A new version of the widevine plugin is available: ${CHROME_VERSION}"
+           rm -f ${WIDEVINE_VERSION_FILE}
```

Comment:
Why remove the version file?  Wouldn't it be better to leave the file alone, so that we keep track of the installed version, in case anything goes wrong during the download/unpack/install?